src/sif/truckcost.cc

```diff
@@ -447,7 +465,7 @@ inline bool TruckCost::Allowed(const baldr::DirectedEdge* edge,
     return false;
   }
 
-  return DynamicCost::EvaluateRestrictions(access_mask_, edge, is_dest, tile, edgeid, current_time,
+  return DynamicCost::EvaluateRestrictions(kTruckAccess, edge, is_dest, tile, edgeid, current_time,
```

Comment:
then we have to pass `kTruckAccess` here explicitly as we still want to evaluate truck restrictions regardless of that costing option